cloudify_aws/ec2/resources/transit_gateway_routetable.py

```diff
@@ -0,0 +1,234 @@
+                         ignore_properties=True)
+@decorators.untag_resources
+def delete(ctx, iface, resource_config, **_):
+    '''Deletes an AWS EC2 Transit Gateway Route Table'''
+    params = \
+        dict() if not resource_config else resource_config.copy()
+    params[ROUTETABLE_ID] = iface.resource_id
+    iface.delete(params)
+
+
+@decorators.aws_resource(EC2TransitGatewayRouteTable, RESOURCE_TYPE)
+def attach(ctx, iface, resource_config, **_):
+    '''Attaches an AWS EC2 Transit Gateway Route Table to a Transit Gateway'''
+    params = dict() if not resource_config else resource_config.copy()
+    route_table_id = params.get(ROUTETABLE_ID, iface.resource_id)
+    transit_gateway_attachment_id = get_attachment_id_from_runtime_props(ctx)
+    gw = utils.find_rel_by_node_type(ctx.instance, TG_TYPE)
+    if not gw:
+        raise NonRecoverableError(
+            'Transit Gateway Route Table must provide a '
+            'relationship to a Transit Gateway.')
+    transit_gateway_id = gw.target.instance.runtime_properties.get(
+        EXTERNAL_RESOURCE_ID)
+    result = iface.describe(
+        {
+            'Filters': [
+                {
+                    'Name': 'transit-gateway-id',
+                    'Values': [transit_gateway_id]
+                },
+                {
+                    'Name': 'transit-gateway-route-table-id',
+                    'Values': [route_table_id]
+                }
+            ]
+        }
+    )
+    if ROUTETABLES in result and len(result[ROUTETABLES]) == 1:
+        if result[ROUTETABLES][0][ROUTETABLE_ID] == route_table_id:
+            return
+        detach_request = {
+            ROUTETABLE_ID: result[ROUTETABLES][ROUTETABLE_ID],
+            TG_ATTACHMENT_ID: transit_gateway_attachment_id
+        }
+        iface.detach(detach_request)
+        raise OperationRetry(
+            'Removing existing route table association: {r}'.format(
+                r=result))
+    request = {
+        ROUTETABLE_ID: route_table_id,
+        TG_ATTACHMENT_ID: transit_gateway_attachment_id
+    }
+    try:
+        iface.attach(request)
+    except (ClientError, NonRecoverableError) as e:
+        raise OperationRetry(e)
+
+
+@decorators.aws_resource(EC2TransitGatewayRouteTable, RESOURCE_TYPE,
+                         ignore_properties=True)
+def detach(ctx, iface, resource_config, **_):
+    '''Detach an AWS EC2 Transit Gateway Route Table from a Transit Gateway'''
+    params = dict() if not resource_config else resource_config.copy()
+    route_table_id = params.get(ROUTETABLE_ID, iface.resource_id)
+    transit_gateway_attachment_id = get_attachment_id_from_runtime_props(ctx)
+    request = {
+        ROUTETABLE_ID: route_table_id,
+        TG_ATTACHMENT_ID: transit_gateway_attachment_id
+    }
+    return utils.exit_on_substring(iface,
+                                   'detach',
+                                   request,
+                                   'InvalidAssociation.NotFound')
+
```

Comment:
If they will change the error message it is going to be a bug. Is there a way not to count on error messages?

Reply:
Not worried about it.